lib/apiservers/engine/backends/backends.go

```diff
@@ -346,18 +347,30 @@ func EventService() *events.Events {
 // RegistryCheck checkes the given url against the registry whitelist, blacklist, and insecure
 // registries lists. It returns true for each list where u matches that list.
 func (d *dynConfig) RegistryCheck(ctx context.Context, u *url.URL) (wl bool, bl bool, insecure bool) {
-	c, err := d.src.Get(ctx)
+	d.Lock()
+	src := d.src
+	d.Unlock()
+
+	c, err := src.Get(ctx)
 
 	d.Lock()
 	defer d.Unlock()
 
 	if err == nil {
 		// update config
-		if err := d.update(c); err != nil {
+		if err = d.update(c); err != nil {
 			log.Warnf("error updating config: %s", err)
 		}
-	} else {
+	}
+
+	if err != nil && err != dynamic.ErrConfigNotModified {
 		log.Warnf("could not get config from remote source: %s", err)
+		if src == d.src {
```

Comment:
So (1.) is going to be impossible to figure out what the specific reason is. For (2.) if the source in unaware of the VCH, we may want to do another discovery on next call to find a source that may _now_ be aware of the VCH, or the existing source may now be aware of the VCH.